src/lib/y2partitioner/widgets/delete_disk_partition_button.rb

```diff
@@ -44,22 +66,71 @@ def handle
 
     private
 
+      # Current device to delete
+      #
+      # @return [Y2Storage::BlkDevice]
+      def device
+        @device || @table.selected_device
+      end
+
+      # Checks whether the device can be deleted
+      #
+      # @note Some popups could be shown when it is not possible
+      #   to delete the device.
+      #
+      # @return [Boolean] {true} if it is possible to delete the device;
+      #   {false} otherwise.
+      def delete_validations
+        presence_validation && not_empty_partitions_validation
+      end
+
+      # Checks whether there is a device to delete
+      #
+      # @note An error popup is shown when there is no device.
+      #
+      # @return [Boolean]
+      def presence_validation
+        return true unless device.nil?
+
+        Yast::Popup.Error(_("No device selected"))
+        false
+      end
+
+      # Checks whether a partitionable has any partition for deleting
+      #
+      # @note An error popup is shown when there is no partition.
+      #
+      # @return [Boolean]
+      def not_empty_partitions_validation
```

Comment:
Hmm, I find the name (and, in general, how this validations are structured) a little bit confusing. What about, for example?

```ruby 
def delete_validations
  if device.is?(:partition)
    validate_presence
  else
    validate_presence && validate_empty_table
  end
end

# This is your #not_empty_partitions_validation, but it will never be called if this is a partition
def validate_empty_table
  partition_table = device.partition_table
  blah, blah
end
```

As a bonus, you wouldn't need to add support for `is?(:partitionable)` which is something that does not makes me specially happy.